packages/docusaurus-plugin-content-docs/src/sidebarItemsGenerator.ts

```diff
@@ -268,10 +268,10 @@ export const DefaultSidebarItemsGenerator: SidebarItemsGenerator = async functio
     }
 
     // async process made sequential on purpose! order matters
-    for (const doc of docs) {
-      // eslint-disable-next-line no-await-in-loop
-      await handleDocItem(doc);
-    }
+    await docs.reduce(
```

Comment:
actually I'd prefer to revert this change

Reply:
Although I reverted this, I'm pretty confident that the new code works correctly😂 Just that it is less readable

Reply:
yes definitively works with reduce but much more complicated :)